### PR TITLE
statistics: introduce a new table `mysql.analyze_jobs_status` to store all analyze jobs' information

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: related to #20141 

Problem Summary: statistics: introduce a new table `mysql.analyze_jobs_status` to store all analyze jobs' information

### What is changed and how it works?

This is the first PR related to #20141, and in this PR:
1. a new table `mysql.analyze_jobs_status` is introduced to store analyze jobs' information;
2. `statistic.Handler` is updated to support updating and readding this table;

### Check List 

Tests 

- Unit test

### Release note 

- statistics: introduce a new table `mysql.analyze_jobs_status` to store all analyze jobs' information

### CODE REVIEW

File: statistics/analyze_jobs.go [hunk 1]
```diff
@@ -90,6 +91,12 @@ func GetAllAnalyzeJobs() []*AnalyzeJob {
 	return jobs
 }
 
+// GetChangedAnalyzeJobs returns all changed analyze jobs after last access.
+func GetChangedAnalyzeJobs() []AnalyzeJob {
+	// TODO: complete this function
```
Comment (qw4990) [inline statistics/analyze_jobs.go:96]: To keep this PR small and easy to review, this function will be completed in the next PR.

File: statistics/analyze_jobs.go [hunk 2]
```diff
@@ -30,6 +30,7 @@ var analyzeStatus = analyzeJobs{jobs: make(map[*AnalyzeJob]struct{}), history: m
 // AnalyzeJob is used to represent the status of one analyze job.
 type AnalyzeJob struct {
 	sync.Mutex
+	UUID          string // used to identify jobs
```
Comment (winoros) [inline statistics/analyze_jobs.go:33]: how will we generate its values?

**Review by eurekaka — COMMENTED** [review]
What are the target scenarios of this feature?

- the behavior that `show analyze status` returns the analyze status of its corresponding instance seems reasonable to me, just like `show processlist`;
- why do we need to dump the status info to a table? what is the value of these info when an `analyze` has finished its exeuction?

**Comment by qw4990** [comment]
> What are the target scenarios of this feature?
> 
> * the behavior that `show analyze status` returns the analyze status of its corresponding instance seems reasonable to me, just like `show processlist`;
> * why do we need to dump the status info to a table? what is the value of these info when an `analyze` has finished its exeuction?

It can help us confirm if some tables are analyzed at a specific time, and can check if the `auto-analyze` feature works as we expect.

File: domain/domain.go [hunk 3]
```diff
@@ -1141,10 +1142,12 @@ func (do *Domain) updateStatsWorker(ctx sessionctx.Context, owner owner.Manager)
 	gcStatsTicker := time.NewTicker(100 * lease)
 	dumpFeedbackTicker := time.NewTicker(200 * lease)
 	loadFeedbackTicker := time.NewTicker(5 * lease)
+	updateAnalyzeJobStatusTicker := time.NewTicker(time.Second * 30)
```
Comment (rebelice) [inline domain/domain.go:1145]: Does `time.Second * 30` need to be a configurable variable?

**Comment by breeswish** [comment]
Should we bring this feature in 5.0?

**Comment by tangenta** [comment]
@qw4990 Why do you close this PR? What is the plan?

**Pull request closed without merging**